### PR TITLE
Move most of repo to RN 0.63

### PR DESCRIPTION
**Pull request by Saadnajmi**
### Platforms Impacted
- [x] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [x] android

### Description of changes

Previously, I tried moving just the iOS / android / macOS test apps to RN 0.63. This caused problems as our monorepo structure meant we would pull in multiple versions of react and react native. That caused problems with the CLI among other things, and I realized we'll need to update the monorepo to only pull one version of React Native globally to solve this.

To achieve this, I did a few things :

#### Changed most direct dependencies on react-native to a peer dependency up to the next major version:
```
// Old
"dependencies": {
  "react": "16.11.0"
  "react-native": "0.62.2"
}
// New
"peerDependencies": {
  "react": "^16.11.0"
  "react-native": "^0.62.2"
}
```
I chose to use peer dependencies after reading this [blog post](https://classic.yarnpkg.com/blog/2018/04/18/dependencies-done-right/) on the Yarn website. Peer dependencies  They explicitly mention react as an example of a dependency well suited to a be a peer dependency. Peer dependencies will not pull in the package if it is not in node_modules, but instead throw an error. This guarantees that only the exact version of the package we specified is imported. We need to do this for both react and react native.

The packages that DO specify react-native as a dependency are:
- the test apps (It makes sense that the test apps specify exactly what version they pull in)
- @fluentui-react-native/adapters (this package seemed to weirdly specifically need to import the packages directly)

#### Set RN 0.62 as the minimum for most packages, and pull in RN 0.63.4 as the default

Many packages set their dependencies as both ` "react-native": ">=0.60.0"` and also `"react-native": "0.62.2"`. I interpreted this as fallout from a previous RN version upgrade and decided to specify RN 0.62 as the minimum.

#### Updated our test apps to RN 0.63 (Except windows)

This was initially done by reverting my previous change https://github.com/microsoft/fluentui-react-native/commit/a4242d64947276d406946197db634a36df46cc28 (which iteslf was reverting my change to add RN 0.63 to iOS and android) and then updating the dependences of the other remaining test apps. 

For web, I didn't really need any changes and still ran fine
For windows, I used nohoist to keep the RN 0.62 dependency and test app as I did not want to figure out how to update the template app and whether there was anything custom done there.

win32 is still the WIP. I will probably need outside help to correctly handle that test app.

#### Remove test pages that don't work on iOS and macOS

I should have just done this a long time ago. A bunch of the test pages have components that need native modules that only exist on windows. Let's just remove them. 

### Verification

iOS, macOS, android, windows and web test apps build and run. win32 does mot.

I ran `yarn why` to check which version of react and react-native were hoisted :
```
saads-macbook-pro:fluentui-react-native sanajmi$ yarn why react
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "react"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "react@16.14.0"
info Reasons this module exists
   - "_project_#@fluentui-react-native#tester-windows" depends on it
   - Hoisted from "_project_#@fluentui-react-native#tester-windows#react"
   - Hoisted from "_project_#@uifabricshared#foundation-composable#react"
   - Hoisted from "_project_#@fluentui-react-native#tester-android#react"
   - Hoisted from "_project_#@fluentui-react-native#tester-ios#react"
   - Hoisted from "_project_#@fluentui-react-native#tester-web#react"
   - Hoisted from "_project_#@fluentui-react-native#tester-win32#react"
info Disk size without dependencies: "332KB"
info Disk size with unique dependencies: "520KB"
info Disk size with transitive dependencies: "600KB"
info Number of shared dependencies: 5
✨  Done in 3.53s.
saads-macbook-pro:fluentui-react-native sanajmi$ yarn why react-native
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "react-native"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "react-native@0.63.4"
info Has been hoisted to "react-native"
info Reasons this module exists
   - "workspace-aggregator-54726485-23a5-40ff-bdca-093e832b8a11" depends on it
   - Hoisted from "_project_#@fluentui-react-native#tester-android#react-native"
   - Hoisted from "_project_#@fluentui-react-native#tester-ios#react-native"
   - Hoisted from "_project_#@fluentui-react-native#tester-macos#react-native"
   - Hoisted from "_project_#@fluentui-react-native#tester-win32#react-native"
info Disk size without dependencies: "39.72MB"
info Disk size with unique dependencies: "127.13MB"
info Disk size with transitive dependencies: "39.72MB"
info Number of shared dependencies: 474
=> Found "@fluentui-react-native/tester-windows#react-native@0.62.2"
info Reasons this module exists
   - "_project_#@fluentui-react-native#tester-windows" depends on it
   - in the nohoist list ["/_project_/@fluentui-react-native/tester-windows/@types/jasmine","/_project_/@fluentui-react-native/tester-windows/react-native","/_project_/@fluentui-react-native/tester-windows/react-native/**"]
info Disk size without dependencies: "38.91MB"
info Disk size with unique dependencies: "131.98MB"
info Disk size with transitive dependencies: "38.91MB"
info Number of shared dependencies: 478
=> Found "@fluentui-react-native/adapters#react-native@0.62.2"
info This module exists because "_project_#@fluentui-react-native#adapters" depends on it.
info Disk size without dependencies: "38.91MB"
info Disk size with unique dependencies: "131.98MB"
info Disk size with transitive dependencies: "38.91MB"
info Number of shared dependencies: 478
✨  Done in 6.04s.
saads-macbook-pro:fluentui-react-native sanajmi$ 
```

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/utils/adapters/src/adapters.win32.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ const _viewMask: IFilterMask<IViewProps> = {
   accessibilityActions: true,
   accessibilityAnnotation: true,
   accessibilityDescribedBy: true,
+  accessibilityDescription: true,
```
Comment (Saadnajmi) [inline packages/utils/adapters/src/adapters.win32.ts:17]: I needed to add this to get this package to build. Hopefully that's fine?

File: apps/windows/package.json [hunk 2]
```diff
@@ -15,21 +15,20 @@
   },
   "dependencies": {
     "@fluentui-react-native/tester": "^0.14.0",
-    "react": "16.11.0",
-    "react-native": "0.62.2",
-    "react-native-windows": "^0.62.1"
+    "react-native": "0.63.4",
+    "react-native-windows": "^0.63.20"
```
Comment (NickGerleman) [inline apps/windows/package.json:19]: I dont think we had anything seriously breaking, but be aware that the template app for RNW can change between major versions.

Comment (NickGerleman) [reply apps/windows/package.json:19]: Same for Android and iOS for that matter.

Would be work taking a look at react-native-upgrade-helper if you have a moment.

File: packages/utils/interactive-hooks/package.json [hunk 3]
```diff
@@ -24,17 +24,15 @@
     "invariant": "^2.2.0"
   },
   "devDependencies": {
-    "@office-iss/react-native-win32": "0.62.6",
     "@types/react": "^16.9.34",
```
Comment (NickGerleman) [inline packages/utils/interactive-hooks/package.json:27]: Don't forget to upgrade typings for react + react-native as well.

**Review by NickGerleman — APPROVED** [review]
Overall LGTM.

File: apps/windows/package.json (hunk 2 above)
Comment (Saadnajmi) [reply apps/windows/package.json:19]: We use [React-Native-Test-App](https://github.com/microsoft/react-native-test-app) for the native template code. I have a PR https://github.com/microsoft/fluentui-react-native/pull/552 to move the windows test app as well to use it, but it's blocked on a RNW CLI change that won't go in till RNW 0.64 I believe.

Where could I find the correct template app? I thought react-native-upgrade-helper only did iOS and android?

File: packages/utils/interactive-hooks/package.json (hunk 3 above)
Comment (Saadnajmi) [reply packages/utils/interactive-hooks/package.json:27]: Right. Those versions should just correspond with the same version of react/react native yes?

File: packages/libraries/core/react-native.config.js [hunk 4]
```diff
@@ -1,3 +0,0 @@
-module.exports = {
-  reactNativePath: '../../node_modules/@office-iss/react-native-win32'
-};
```
Comment (Saadnajmi) [inline packages/libraries/core/react-native.config.js:3]: I also wanted to point out I deleted this... it feels weird to me a cross platform framework would require react-native-win32 as it's react native version

**Comment by Saadnajmi** [comment]
As discussed offline, I'm closing this PR for now. It grew more monolithic than I would like and I'd rather not introduce breaking changes.
For future reference, separate PR's that could be implemented in preparation fr 0.63:
- Changing all RN dependencies to peer dependencies, and adding a single resolution at the root package.json
- (Small) fix to our fork of Pressability to fix a type error that happens with 0.63
- Remove some explicit references to react-native-win32 in some of the base components and move that to a *.win32.tsx file

I'll try to do some of those in the next fork, but out of time for now!

**Pull request closed without merging**